plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/project/PrefPageProjectNavigatorView.java

```diff
@@ -0,0 +1,211 @@
+            container,
+            UINavigatorMessages.pref_page_navigator_view_group_settings,
+            2, GridData.FILL_HORIZONTAL, 0
+        );
+
+        UIUtils.createControlLabel(group, UINavigatorMessages.pref_page_navigator_view_label_connection_view);
+        combo = new Combo(group, SWT.DROP_DOWN | SWT.READ_ONLY);
+        combo.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
+
+        combo.add(DataSourceNavigatorSettings.PRESET_SIMPLE.getName());
+        combo.add(DataSourceNavigatorSettings.PRESET_FULL.getName());
+        combo.add(DataSourceNavigatorSettings.PRESET_CUSTOM.getName());
+
+        loadValues();
+
+        if (DBWorkbench.isDistributed() && !currentUserIsAdmin()) {
+            combo.setEnabled(false);
+            setMessage(UINavigatorMessages.pref_page_navigator_view_message_admin_only, WARNING);
+        }
+
+        return container;
+    }
+
+    private boolean currentUserIsAdmin() {
+        return DBWorkbench.getPlatform().getWorkspace().hasRealmPermission(DBAPermissionRealm.PERMISSION_ADMIN);
+    }
+
+    private void loadValues() {
+        if (projectMeta == null) {
+            combo.select(0);
+            return;
+        }
+
+        var dataSources = projectMeta.getDataSourceRegistry().getDataSources();
+        if (dataSources.isEmpty()) {
+            combo.select(0);
+            return;
+        }
+
+        DBNBrowseSettings firstSettings = dataSources.getFirst().getNavigatorSettings();
+        boolean allSame = true;
+
+        for (int i = 1; i < dataSources.size(); i++) {
+            DBNBrowseSettings currentSettings = dataSources.get(i).getNavigatorSettings();
+            if (!firstSettings.equals(currentSettings)) {
+                allSame = false;
+                break;
+            }
+        }
+
+        int selectedIndex = 0;
+
+        if (!allSame) {
+            selectedIndex = PRESETS.get(DataSourceNavigatorSettings.PRESET_CUSTOM);
+        } else {
+            for (var preset : DataSourceNavigatorSettings.PRESETS.values()) {
+                if (preset.getSettings().equals(firstSettings)) {
+                    selectedIndex = PRESETS.get(preset);
+                    break;
+                }
+            }
+        }
+        combo.select(selectedIndex);
+    }
+
+    @Override
+    public boolean performOk() {
+        if (projectMeta == null || combo == null) {
+            return super.performOk();
+        }
+
+        String selectedValue;
+        int selectedIndex = combo.getSelectionIndex();
+
+        if (selectedIndex == PRESETS.get(DataSourceNavigatorSettings.PRESET_SIMPLE)) {
+            selectedValue = DataSourceNavigatorSettings.PRESET_SIMPLE.getId();
+        } else if (selectedIndex == PRESETS.get(DataSourceNavigatorSettings.PRESET_FULL)) {
+            selectedValue = DataSourceNavigatorSettings.PRESET_FULL.getId();
+        } else {
+            selectedValue = DataSourceNavigatorSettings.PRESET_CUSTOM.getId();
+        }
```

Comment:
Following the suggested approach, you could do `PRESETS.get(selectedIndex).getId()`

Reply:
fixed